python/gen_requirements.py

```diff
@@ -231,7 +231,7 @@
     ("sphinx_autodoc_annotation", None),
     ("sphinx_gallery", None),
     ("sphinx_rtd_theme", None),
-    ("synr", ">=0.2.1"),  # Requires bugfix commit ee0b12a61c08f01604475f36ff37d4cb110bdc27
+    ("synr", "==0.3.0"),
```

Comment:
can you comment "depends on this precise version across containers?" or something

Reply:
I'm not sure on what is the reason to be pinned in this specific version. Do you know @jroesch ?

Reply:
I think `synr` was being rapidly developed to work around Python AST stability and there was some reason that @tkonolige and @altanh left it pinned in some places.

Reply:
I would pin it to `0.3` so you can get non-breaking fixes as minor version bumps. Currently TVM is tested against and works with 0.3.0, but 0.3.1 is a non-breaking improvement.

Reply:
@tkonolige I didn't feel very comfortable to do anything different from what is done in `task_ci_setup.sh`.

https://github.com/apache/tvm/blob/e691c7f83892d7242d0992c78cec1e2f8953a9e3/tests/scripts/task_ci_setup.sh#L33

As I'm just reading this comment now, and the PR is already merged, feel free to send a follow-up PR.